buildbot_nix/repo_config/__init__.py

```diff
@@ -0,0 +1,42 @@
+import tomllib
+from tomllib import TOMLDecodeError
+from typing import TYPE_CHECKING, Self
+
+from buildbot.process.buildstep import ShellMixin
+
+if TYPE_CHECKING:
+    from buildbot.process.log import Log
+from pydantic import BaseModel, ValidationError
+
+
+class RepoConfig(BaseModel):
+    branches: list[str]
+
+
+class BranchConfig(BaseModel):
+    lock_file: str = "flake.lock"
+    attribute: str = "checks"
+
+    @classmethod
+    async def extract_during_step(cls, buildstep: ShellMixin) -> Self:
+        stdio: Log = await buildstep.addLog("stdio")
+        cmd = await buildstep.makeRemoteShellCommand(
+            collectStdout=True,
+            collectStderr=False,
+            stdioLogName=None,
+            # TODO: replace this with something like buildbot.steps.transfer.StringUpload
+            # in the future... this one doesn't not exist yet.
+            command=["cat", "buildbot-nix.toml"],
+        )
+        await buildstep.runCommand(cmd)
+        if cmd.didFail():
+            stdio.addStdout("Failed to read repository local configuration.\n")
```

Comment:
If configuration does not exist, I think we should not show any error and have it optional instead.